### PR TITLE
[SYCL-MLIR] Define `sycl.host.handler.set_nd_range`

### PR DESCRIPTION
**Pull request by victor-eds**
Operation assigning an nd-range to a `sycl::handler`, setting the nd-range of the kernel to be launched.

The input nd-range can be specified either with a memref to a `sycl.nd_range` or a memref to a `sycl.range` (global size) and an optional memref to a `sycl.id` (offset).

### CODE REVIEW

**Comment by victor-eds** [comment]
Depends on #10161

File: mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp [hunk 1]
```diff
@@ -388,5 +395,29 @@ SYCLHostHandlerSetKernel::verifySymbolUses(SymbolTableCollection &symbolTable) {
...
+  case 2:
+    if (auto globalSizeType = getMemRefElementType<RangeType>(ndRangeType[0])) {
+      if (auto offsetType = getMemRefElementType<IDType>(ndRangeType[1])) {
+        return getDimensions(globalSizeType) == getDimensions(offsetType)
+                   ? success()
+                   : emitOpError("expects both global size and offset to have "
+                                 "the same number of dimensions");
+      }
+    }
+    [[fallthrough]];
```
Comment (jopperm) [inline mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp:415]: Why not just `break;`?

File: mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp [hunk 2]
```diff
@@ -44,6 +44,13 @@ static LogicalResult checkSameDimensions(Operation *op, Type thisType,
                    << inputDims << " vs " << thisDims;
 }
 
+template <typename Ty> static Ty getMemRefElementType(Type ty) {
+  auto mt = dyn_cast<MemRefType>(ty);
+  if (!mt)
+    return {};
+  return dyn_cast<Ty>(mt.getElementType());
```
Comment (jopperm) [inline mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp:51]: Nit / matter of taste:
```suggestion
return mt ? dyn_cast<Ty>(mt.getElementType()) : {};
```

**Review by jopperm — APPROVED** [review]

File: mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp (hunk 2 above)
Comment (victor-eds) [reply mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp]: This was part of a different PR. Now that I rebased it's gone

**Review by sommerlukas — APPROVED** [review]

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**